internal/hmyapi/apiv2/blockchain.go

```diff
@@ -65,7 +65,7 @@ func (s *PublicBlockChainAPI) isBeaconShard() error {
 }
 
 func (s *PublicBlockChainAPI) isBlockGreaterThanLatest(blockNum uint64) error {
-	if blockNum > s.b.CurrentBlock().NumberU64() {
+	if blockNum > (s.b.CurrentBlock().NumberU64() + 1) {
```

Comment:
+1 needed? I think no change is needed.

Reply:
Requesting current block + 1 is valid in some cases.

Reply:
I am not comfortable allowing blockNr+1, unless all callsites of `isBlockGreaterThanLatest` have proper checks for block to be nil when extracted using `BlockByNumber`. For example, `GetDelegationsByDelegatorByBlockNumber` if the block does not exists, 
`block, err := s.b.BlockByNumber(ctx, rpc.BlockNumber(blockNum))` will not return any error instead block is nil, but you check
`if err != nil {`, which will not stop a nil block from getting past this check and crash with nil pointer somewhere down. Can you please fix all the callsites for such potential nil block situations? Also, why is allowing blockNr+1 necessary?